src/components/questions/addAnswer.js

```diff
@@ -0,0 +1,147 @@
+import React from "react";
+import axios from "axios";
+
+class AddAnswer extends React.Component {
+  constructor(props){
+    super(props);
+    this.state = {
+      qID : this.props.qID,
+      answer:"",
+      nickname:"",
+      email:"",
+      requiredAnswer: false,
+      requiredNickname: false,
+      requiredEmail: false,
+      validEmail: false
+    }
+    this.onChange = this.onChange.bind(this);
+    this.submitAnswer = this.submitAnswer.bind(this);
+    this.checkEmailValidity = this.checkEmailValidity.bind(this);
+  }
+
+  onChange(event){
+    console.log(event.target.value);
+    let feild = event.target.name;
+    let value = event.target.value;
+    this.setState({[feild] : value})
+  }
+
+  checkEmailValidity(email){
+    let regex = RegExp(/^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$/);
+    return regex.test(email)
+  }
+
+  submitAnswer(event){
+    // event.preventDefault()
+    if (this.state.answer.length === 0) {
```

Comment:
Maybe think about pulling out this out into a separate function to modularize the code in this file a little more. Great concept though.